java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java

```diff
@@ -752,8 +752,14 @@ public void setBlob(int parameterIndex, Blob x) throws SQLException {
     }
 
     public void setClob(int parameterIndex, Clob x) throws SQLException {
-        throw new SQLFeatureNotSupportedException(
-            Constants.SQLExceptionMessages.SQL_FEATURE_NOT_SUPPORTED);
+        checkOpen();
+        if (x.length() > Integer.MAX_VALUE) {
+            throw new SQLFeatureNotSupportedException(
```

Comment:
Is this the right error.
Is there a way to implement this feature when x.length() > Integer.MAX_VALUE?